File: docs/WidgetMill/packages/spec/src/nodes/column.ts
import { z } from "zod";
import { NodeRef } from "../node-ref";
import { flexProps } from "./_flex";

/** Column — spec §2.2. */
export const ColumnNode = z.object({
  type: z.literal("column"),
  props: flexProps.default({}),
  children: z.array(NodeRef).default([]),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/column.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Column — spec §2.2.
Widget buildColumn(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Column(
    mainAxisAlignment: mainAxisAlignmentFrom(p['mainAxisAlignment']),
    crossAxisAlignment: crossAxisAlignmentFrom(p['crossAxisAlignment']),
    mainAxisSize: mainAxisSizeFrom(p['mainAxisSize']),
    spacing: parseDouble(p['spacing']) ?? 0,
    children: r.renderAll(context, node.children),
  );
}
